tests/remoteInstances/remoteInstancesHelper.js

```diff
@@ -0,0 +1,212 @@
+      },
+    },
+    haproxy: {
+      haproxy_2: {
+        host: '192.168.0.1',
+        port: '42100',
+        clusterName: 'haproxy_clst',
+      },
+    },
+    external: {
+      redis: {
+        host: '192.168.0.1',
+        port: '42200',
+        clusterName: 'redis_external_exporter',
+      },
+    },
+    aws: {
+      aws_access_key: process.env.AWS_ACCESS_KEY_ID,
+      aws_secret_key: process.env.AWS_SECRET_ACCESS_KEY,
+      aws_rds_5_7: {
+        address: process.env.REMOTE_AWS_MYSQL57_HOST,
+        username: process.env.REMOTE_AWS_MYSQL_USER,
+        password: process.env.REMOTE_AWS_MYSQL_PASSWORD,
+        clusterName: 'aws_rds_mysql_5_7',
+        port: 3306,
+      },
+      aws_rds_5_6: {
+        address: process.env.REMOTE_AWS_MYSQL57_HOST,
+        username: process.env.REMOTE_AWS_MYSQL_USER,
+        password: process.env.REMOTE_AWS_MYSQL_PASSWORD,
+        clusterName: 'aws_rds_mysql_5_6',
+        port: 3306,
+      },
+      aws_postgresql_12: {
+        userName: process.env.REMOTE_AWS_POSTGRES12_USER,
+        password: process.env.REMOTE_AWS_POSTGRES12_PASSWORD,
+        clusterName: 'aws_postgresql_12',
+        port: 5432,
+      },
+    },
+    azure: {
+      azure_client_id: process.env.AZURE_CLIENT_ID,
+      azure_client_secret: process.env.AZURE_CLIENT_SECRET,
+      azure_tenant_id: process.env.AZURE_TENNANT_ID,
+      azure_subscription_id: process.env.AZURE_SUBSCRIPTION_ID,
+      azure_mysql: {
+        userName: process.env.AZURE_MYSQL_USER,
+        password: process.env.AZURE_MYSQL_PASS,
+      },
+      azure_postgresql: {
+        userName: process.env.AZURE_POSTGRES_USER,
+        password: process.env.AZURE_POSTGRES_PASS,
+      },
+    },
+  },
+
+  instanceTypes: {
+    mysql: (remoteInstanceStatus.mysql.ps_5_7.enabled ? 'MySQL' : undefined),
+    postgresql: (remoteInstanceStatus.postgresql.pdpgsql_13_3.enabled ? 'PostgreSQL' : undefined),
+    mongodb: (remoteInstanceStatus.mongodb.psmdb_4_2.enabled ? 'MongoDB' : undefined),
+    proxysql: (remoteInstanceStatus.proxysql.proxysql_2_1_1.enabled ? 'ProxySQL' : undefined),
+    rds: 'RDS',
+  },
+
+  serviceTypes: {
+    mysql: (
+      remoteInstanceStatus.mysql.ps_5_7.enabled ? {
+        serviceType: 'MYSQL_SERVICE',
+        service: 'mysql',
+      } : undefined
+    ),
+    mongodb: (
+      remoteInstanceStatus.mongodb.psmdb_4_2.enabled ? {
+        serviceType: 'MONGODB_SERVICE',
+        service: 'mongodb',
+      } : undefined
+    ),
+    postgresql: (
+      remoteInstanceStatus.postgresql.pdpgsql_13_3.enabled ? {
+        serviceType: 'POSTGRESQL_SERVICE',
+        service: 'postgresql',
+      } : undefined
+    ),
+    proxysql: (
+      remoteInstanceStatus.proxysql.proxysql_2_1_1.enabled ? {
+        serviceType: 'PROXYSQL_SERVICE',
+        service: 'proxysql',
+      } : undefined
+    ),
+  },
+
+  services: {
+    mysql: (remoteInstanceStatus.mysql.ps_5_7.enabled ? 'mysql_remote_new' : undefined),
```

Comment:
why do we need to set undefined? and how do we handle it in case the value is undefined?

Reply:
while preparing data tables we need to check if a service key value is undefined, we will exclude it.